board/safety/safety_subaru.h

```diff
@@ -40,15 +45,29 @@ static void subaru_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
       controls_allowed = 0;
     }
     subaru_cruise_engaged_last = cruise_engaged;
+  } else if ((addr == 0x144) && (bus_number == 0)) {
+    int cruise_engaged = (to_push->RDHR >> 17) & 1;
+    if (cruise_engaged && !subaru_cruise_engaged_last) {
```

Comment:
this should also be combined with ^^. 
Let me refactor it. I'll open a new PR. @bugsy924